### PR TITLE
Fix #1073, resolve names from def macro expandees.

### LINKED ISSUE #1073: Def macro calls don't resolve to symbols
```scala
package fix

import org.scalatest._
object foo extends FunSuite {
  test("blah") {
    assert(1 == 2)
  }
}

```

has the following smdb

```scala
Language:
Scala212

Names:
[8..11): fix <= _root_.fix.
[20..23): org => _root_.org.
[24..33): scalatest => _root_.org.scalatest.
[43..46): foo <= _root_.fix.foo.
[55..63): FunSuite => _root_.org.scalatest.FunSuite#
[68..72): test => _root_.org.scalatest.FunSuiteLike#test(Ljava/lang/String;Lscala/collection/Seq;Lscala/Function0;Lorg/scalactic/source/Position;)V.
[96..98): == => _root_.scala.Int#`==`(I)Z.

Symbols:
_root_.fix. => package fix
_root_.fix.foo. => final object foo
_root_.org. => package org
_root_.org.scalatest. => package scalatest
_root_.org.scalatest.FunSuite# => class FunSuite
_root_.org.scalatest.FunSuite#`<init>`()V. => primaryctor <init>: ()org.scalatest.FunSuite
_root_.org.scalatest.FunSuiteLike#test(Ljava/lang/String;Lscala/collection/Seq;Lscala/Function0;Lorg/scalactic/source/Position;)V. => protected def test: (testName: String, testTags: org.scalatest.Tag*)(testFun: => Any)(implicit pos: org.scalactic.source.Position)Unit
_root_.scala.Int#`==`(I)Z. => abstract def ==: (x: Int)Boolean

Sugars:
[105..105): *(org.scalactic.source.Position.apply("Fix.scala", "/Users/ollie/dev/scalafix/scalafix-tests/input/src/main/scala/fix/Fix.scala", 5))
  [0..1): * => _star_.
  [32..37): apply => _root_.org.scalactic.source.Position.apply(Ljava/lang/String;Ljava/lang/String;I)Lorg/scalactic/source/Position;.
```
observe that `assert` does not resolve to a symbols. It would be valuable for some analysis to know if a name resolves to a macro.

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -343,6 +343,7 @@ lazy val semanticdbIntegration = project
       )
     }
   )
+  .dependsOn(scalametaJVM)
```
Comment (xeno-by) [inline build.sbt]: Try using `classTag[...]`

File: scalameta/semanticdb-integration/src/main/scala/example/Example.scala [hunk 2]
```diff
@@ -7,4 +7,5 @@ object Example {
   def main(args: Array[String]): Unit = {
     println(1)
   }
+  val x = sourcecode.Name.generate
```
Comment (xeno-by) [inline scalameta/semanticdb-integration/src/main/scala/example/Example.scala]: See above

File: build.sbt (hunk 1 above)
Comment (xeno-by) [inline build.sbt]: Or depend on sourcecode directly.

Comment (olafurpg) [reply build.sbt]: Done, used classTag

**Pull request merged**